compat/getaddrinfo.cc

```diff
@@ -262,7 +258,7 @@ xgetaddrinfo (const char *nodename, const char *servname,
         }
 
         result.ai_family = hp->h_addrtype;
-        ai = dup_addrinfo (&result, &sa, addrlen);
+        struct addrinfo *ai = dup_addrinfo(&result, &sa, addrlen);
```

Comment:
Please simplify and improve const-correctness if possible :

```suggestion
        const auto ai = dup_addrinfo(&result, &sa, addrlen);
```